tests/unit/future/test_polling.py

```diff
@@ -108,7 +110,7 @@ def test_result_with_polling():
 
 
 class PollingFutureImplTimeout(PollingFutureImplWithPoll):
```

Comment:
Let's add a `PollingFutureImplWithoutRetry` and related tests to ensure in unit tests that we aren't breaking older clients.